practice-app/client/pages/location/map.tsx

```diff
@@ -0,0 +1,176 @@
+import { MapContainer, Marker, Popup, TileLayer,useMapEvents } from 'react-leaflet'
+import 'leaflet/dist/leaflet.css'
+import * as L from 'leaflet';
+import { useState } from 'react';
+import iconMarker from 'leaflet/dist/images/marker-icon.png'
+import iconRetina from 'leaflet/dist/images/marker-icon.png'
+import iconShadow from 'leaflet/dist/images/marker-shadow.png'
+import axios from 'axios';
+
+
+var greenIcon = new L.Icon({
+  iconUrl: 'https://raw.githubusercontent.com/pointhi/leaflet-color-markers/master/img/marker-icon-2x-green.png',
+  shadowUrl: 'https://cdnjs.cloudflare.com/ajax/libs/leaflet/0.7.7/images/marker-shadow.png',
+  iconSize: [25, 41],
+  iconAnchor: [12, 41],
+  popupAnchor: [1, -34],
+  shadowSize: [41, 41]
+});
+
+var redIcon = new L.Icon({
+  iconUrl: 'https://raw.githubusercontent.com/pointhi/leaflet-color-markers/master/img/marker-icon-2x-red.png',
+  shadowUrl: 'https://cdnjs.cloudflare.com/ajax/libs/leaflet/0.7.7/images/marker-shadow.png',
+  iconSize: [25, 41],
+  iconAnchor: [12, 41],
+  popupAnchor: [1, -34],
+  shadowSize: [41, 41]
+});
+
+const Map = () => {
+  const [MarkerArr, setMarkerArr] = useState([]);
+  const [selectedPosition, setSelectedPosition] = useState({x_coord: "" ,y_coord:""});
+  //{x_coord: "" ,y_coord:""}
+  const [responseData, setResponseData] = useState(null);
+  const MyFetchButton = () => {
+
+    const handleClickFetch = () => {
+
+      // Send a POST request to your API
+      axios.get('http://127.0.0.1:8000/location/')
+        .then(response => {
+          // Handle the response from the API
+          console.log('API response:', response.data);
+          setResponseData(response.data);
+          
+        })
+        .catch(error => {
+          // Handle any error that occurred during the request
+          console.error('Errror:', error);
+        });
+    };
+
+    return (
+      <div>
+        <button onClick={handleClickFetch}>Collect Data</button>
+        
+      </div>
+    );
+  };
+  const MyInsertButton = () => {
+  
+    const handleClickInsert = () => {
+
+      // Send a POST request to your API
+      console.log(MarkerArr)
+      MarkerArr[0].x_coord = selectedPosition.x_coord;
+      MarkerArr[0].y_coord = selectedPosition.y_coord;
+      
+
+      axios.post('http://127.0.0.1:8000/location/insert',JSON.stringify({items: MarkerArr}),{
```

Comment:
can you write as process.env.NEXT_PUBLIC_BACKEND_URL+'location/insert

Reply:
did it